### PR TITLE
selinux: cache enforced status and treat retrieve failure as enforced mode

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: src/basic/selinux-util.c [hunk 1]
```diff
@@ -49,19 +51,46 @@ bool mac_selinux_use(void) {
 #endif
 }
 
+bool mac_selinux_enforced(void) {
+#if HAVE_SELINUX
+        if (cached_enforced < 0) {
+                cached_enforced = security_getenforce();
+                if (cached_enforced == -1) {
+                        log_error_errno(errno, "Failed to get SELinux enforced status: %m");
+                }
```
Comment (keszybz) [inline src/basic/selinux-util.c:60]: {} not necessary.

File: src/basic/selinux-util.c [hunk 2]
```diff
@@ -32,10 +33,11 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(context_t, context_free);
 #define _cleanup_context_free_ _cleanup_(context_freep)
 
 static int cached_use = -1;
+static int cached_enforced = -1;
 static struct selabel_handle *label_hnd = NULL;
 
-#define log_enforcing(...) log_full(security_getenforce() == 1 ? LOG_ERR : LOG_WARNING, __VA_ARGS__)
-#define log_enforcing_errno(r, ...) log_full_errno(security_getenforce() == 1 ? LOG_ERR : LOG_WARNING, r, __VA_ARGS__)
+#define log_enforcing(...) log_full(mac_selinux_enforced() ? LOG_ERR : LOG_WARNING, __VA_ARGS__)
```
Comment (keszybz) [inline src/basic/selinux-util.c:39]: Can we please call this `mac_selinux_enforcing`? We already have `log_enforcing`, and `_getenforce` and `enforcing=` on the kernel command line, and yet another spelling makes it harder to remember how to spell the thing.

File: src/basic/selinux-util.c [hunk 3]
```diff
@@ -49,19 +51,46 @@ bool mac_selinux_use(void) {
 #endif
 }
 
+bool mac_selinux_enforced(void) {
+#if HAVE_SELINUX
+        if (cached_enforced < 0) {
```
Comment (keszybz) [inline src/basic/selinux-util.c:56]: And `cached_enforcing` here.

**Comment by keszybz** [comment]
I guess this should go in after next release though.

**Comment by cgzones** [comment]
see #15010

**Pull request closed without merging**